sml_small/editing/thousand_pounds/thousand_pounds.py

```diff
@@ -76,24 +119,50 @@ def run(
     ) as error:  # Catch any underlying errors and return a coherent output dataset
         # Ensure we populate the output target variables with the same output values as originally given
         target_variables_final = []
-        for question in target_variables:
-            target_variables_final.append(
-                Target_variable(
-                    identifier=question.identifier,
-                    original_value=question.original_value,
-                    final_value=question.original_value,
-                )
-            )
-
-        return Thousands_output(
-            principal_identifier=principal_identifier,
-            principal_original_value=principal_variable,
-            principal_final_value=principal_variable,  # Always return the final output as the same as the input
-            target_variables=target_variables_final,
-            tpc_ratio=None,
-            tpc_marker="E",
-            error_description=f"{error}",
-        )
+        if principal_identifier is None:
+            principal_identifier = "N/A"
+        raise TPException(f"identifier: {principal_identifier}", error)
+
+
+def validate_input(
+    predictive,
+    auxiliary,
+    principal_variable,
+    lower_limit,
+    upper_limit,
+    target_variables,
+):
+    if predictive and validate_number("predictive", predictive):
+        predictive = float(predictive)
+    if auxiliary and validate_number("auxiliary", auxiliary):
+        auxiliary = float(auxiliary)
+    if predictive is None and auxiliary is None:
+        raise ValueError(get_one_of_params_mandatory_error(["predictive", "auxiliary"]))
+    if principal_variable is None:
+        raise ValueError(get_mandatory_param_error("principal_variable"))
+    elif validate_number("principal_variable", principal_variable) is True:
+        principal_variable = float(principal_variable)
+    if not lower_limit:
+        raise ValueError(get_mandatory_param_error("lower_limit"))
+    elif validate_number("lower_limit", lower_limit) is True:
+        lower_limit = float(lower_limit)
+    if not upper_limit:
+        raise ValueError(get_mandatory_param_error("upper_limit"))
+    elif validate_number("upper_limit", upper_limit) is True:
+        upper_limit = float(upper_limit)
+    if float(lower_limit) > float(upper_limit):
```

Comment:
Can you extend this to be >= as Ellie has provided some info that she wants to capture that scenario too.

"2.	David built in a validation to throw an exception if the lower limit was greater than the upper limit. However, I think needs to be extended so that lower cannot equal upper. The reasoning is explained below the cases in the UAT document."

Reply:
Amended, test added in test_validation to confirm functionality